internal/tool/tool.go

```diff
@@ -0,0 +1,183 @@
+package tool
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v6"
+	"github.com/samber/lo"
+)
+
+const sourceConfigFileName = ".semgrep.yaml"
+
+// New creates a new instance of Codacy Semgrep.
+func New() codacySemgrep {
+	return codacySemgrep{}
+}
+
+// Codacy Semgrep tool implementation
+type codacySemgrep struct {
+}
+
+// https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance
+var _ codacy.Tool = (*codacySemgrep)(nil)
+
+// Run runs the Semgrep implementation
+func (s codacySemgrep) Run(ctx context.Context, toolExecution codacy.ToolExecution) ([]codacy.Result, error) {
+	configFile, err := getConfigurationFile(*toolExecution.Patterns, toolExecution.SourceDir)
+	if err == nil {
+		defer os.Remove(configFile.Name())
+	}
+
+	// filesToAnalyse, err := getListOfFilesToAnalyse(tool.Files, sourceDir)
```

Comment:
not sure why this is needed 🤷 
perhaps this logic predates the golang seed

Reply:
see https://codacy.atlassian.net/browse/TAROT-2433 why it's need
can be addressed in a follow up PR